src/sentry/replays/consumers/replay_payload_consumer.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import annotations
+
+import logging
+from datetime import datetime
+from typing import Any, Dict, Sequence
+
+import msgpack
+from confluent_kafka import Message
+
+from sentry import replaystore
+from sentry.attachments import MissingAttachmentChunks, attachment_cache
+from sentry.ingest.ingest_consumer import trace_func
+from sentry.utils import metrics
+from sentry.utils.batching_kafka_consumer import AbstractBatchWorker, BatchingKafkaConsumer
+from sentry.utils.kafka import create_batching_kafka_consumer
+
+CACHE_TIMEOUT = 3600
+logger = logging.getLogger("sentry.replays")
+
+
+def get_replay_payloads_consumer(
+    topic: str,
+    **options: Dict[str, str],
+) -> BatchingKafkaConsumer:
+    return create_batching_kafka_consumer(
+        {topic},
+        worker=ReplaysPayloadConsumer(),
+        **options,
+    )
+
+
+class ReplaysPayloadConsumer(AbstractBatchWorker):  # type: ignore
```

Comment:
The AbstractBatchWorker will be deprecated (not immediately since it's still in use but eventually). Please consider writing a streaming consumer using the arroyo library instead. You don't really need to wait for batches to fill up for this use case, since the messages get produced one at a time anyway. In the future, it will also make it easier to switch to use multiprocessing if needed.

Reply:
i'll have some spare moments once the rest of my snuba changes are up and i'll take a crack at switching them to arroyo.